pkg/controller/petset/pet_set_test.go

```diff
@@ -263,3 +263,12 @@ func TestPetSetBlockingPetIsCleared(t *testing.T) {
 		t.Errorf("Unexpected blocking pet, err %v: %+v", err, p)
 	}
 }
+
+func TestPetSetReplicaCount(t *testing.T) {
```

Comment:
This test requires me to add a lot of code (mock KubeClient for petset package, so I will do it in another PR)